frontend/src/pages/SubmitProperty.tsx

```diff
@@ -29,27 +36,37 @@ const SubmitProperty = (props: any) => {
   const activeParcelDetail = useSelector<RootState, IParcelDetail>(
     state => state.parcel.parcelDetail as IParcelDetail,
   );
+  const [cachedParcelDetail, setCachedParcelDetail] = useState(activeParcelDetail?.parcelDetail);
+
   const properties = useSelector<RootState, IProperty[]>(state => state.parcel.parcels);
   const dispatch = useDispatch();
-
   if (
-    activeParcelDetail?.parcelDetail?.agencyId &&
-    !keycloak.hasAgency(activeParcelDetail?.parcelDetail?.agencyId)
+    cachedParcelDetail?.agencyId &&
+    !formDisabled &&
+    !keycloak.hasAgency(cachedParcelDetail?.agencyId)
   ) {
-    parsed.disabled = 'true'; //if the user doesn't belong to this properties agency, display a read only view.
+    setFormDisabled(true); //if the user doesn't belong to this properties agency, display a read only view.
   }
-  if (!keycloak.agencyId && !parsed.disabled) {
+  if (!keycloak.agencyId && !formDisabled) {
     throw Error('You must belong to an agency to submit properties');
   } else if (!keycloak.obj?.subject) {
     throw Error('Keycloak subject missing');
```

Comment:
This error probably isn’t helpful if visible to a user.

Reply:
I'm open to suggestions for a change in wording.